File: SkaiLab.Invoice/ClientApp/src/app/models/quote.ts
import { Currency } from "./currency";
import { Customer } from "./customer";
import { Product } from "./product";
import { Tax } from "./tax";
import {Location} from "./location";
import { QuoteEnum } from "./enum";
import { Attachment } from "./attachment";
export class QuoteItem{
    id:number;
    productId:number;
    quantity:number;
    taxId?:number;
    lineTotal:number;
    lineTotalIncludeTax:number;
    product:Product;
    tax:Tax;
    discountRate?:number;
    unitPrice:number;
    locationId:number;
    description:string;
    constructor(){
        this.id=0;
        this.productId=null;
        this.product=new Product();
        this.quantity=null;
        this.taxId=null;
        this.discountRate=null;
        this.tax=null;
        this.lineTotal=0;
        this.lineTotalIncludeTax=0;
        this.unitPrice=0;
        this.locationId=null;
        this.description="";
    }
}
export class QuoteStatus{
    id:number;
    name:string;
    count:number;
    constructor(){
        this.id=0;
        this.name="";
        this.count=0;
    }
}
export class Quote{
    id:number;
    customerId:number;
    organisationId:string;
    created:Date;
    expireDate:Date;
    refNo:string;
    number:string;
    note:string;
    currencyId:number;
    taxCurrencyExchangeRate:number;
    baseCurrencyExchangeRate:number;
    total:number;
    totalIncludeTax:number;
    statusId:number;
    createdBy:string;
    acceptedBy:string;
    acceptedDate:Date;
    declinedBy:string;
    declinedDate:string;
    invoicedBy:string;
    invoicedDate:Date;
    currency:Currency;
    customer:Customer;
    status:QuoteStatus;
    date:Date;
    selected:boolean;
    quoteItems:QuoteItem[];
    attachments:Attachment[];
    termAndCondition:string;
    constructor(){
        this.id=0;
        this.customerId=null;
        this.customer=null;
        this.organisationId="";
        this.created=new Date();
        this.expireDate=null;
        this.refNo="";
        this.number="";
        this.note="";
        this.currencyId=null;
        this.taxCurrencyExchangeRate=0;
        this.baseCurrencyExchangeRate=0;
        this.total=0;
        this.totalIncludeTax=0;
        this.statusId=QuoteEnum.Draft;
        this.createdBy="";
        this.acceptedBy="";
        this.acceptedDate=null;
        this.declinedBy="";
        this.declinedDate=null;
        this.invoicedBy="";
        this.invoicedDate=null;
        this.currency=new Currency();
        this.status=new QuoteStatus();
        this.date=new Date();
        this.quoteItems=[];
        this.attachments=[];
        this.termAndCondition="";
    }
}
export class QuoteForUpdateOrCreate{
    quote:Quote;
    currencies:Currency[];
    locations:Location[];
    taxes:Tax[];
    baseCurrencyId:number;
    taxCurrency:Currency;
    number:string;
    constructor(){
        this.quote=new Quote();
        this.currencies=[];
        this.locations=[];
        this.taxes=[];
        this.taxCurrency=new Currency();
        this.number="";
    }
}

File: SkaiLab.Invoice.Models/Quote.cs
using System;
using System.Collections.Generic;

namespace SkaiLab.Invoice.Models
{
    public class Quote
    {
        public long Id { get; set; }
        public long CustomerId { get; set; }
        public string OrganisationId { get; set; }
        public DateTime Created { get; set; }
        public DateTime Date { get; set; }
        public DateTime? ExpireDate { get; set; }
        public string RefNo { get; set; }
        public string Number { get; set; }
        public string Note { get; set; }
        public int CurrencyId { get; set; }
        public decimal? TaxCurrencyExchangeRate { get; set; }
        public decimal? BaseCurrencyExchangeRate { get; set; }
        public decimal Total { get; set; }
        public decimal TotalIncludeTax { get; set; }
        public int StatusId { get; set; }
        public string CreatedBy { get; set; }
        public string AcceptedBy { get; set; }
        public DateTime? AcceptedDate { get; set; }
        public string DeclinedBy { get; set; }
        public DateTime? DeclinedDate { get; set; }
        public string InvoicedBy { get; set; }
        public DateTime? InvoicedDate { get; set; }

        public virtual Currency Currency { get; set; }
        public virtual Customer Customer { get; set; }
        public virtual QuoteStatus Status { get; set; }
        public List<QuoteItem> QuoteItems { get; set; }
        public List<Attachment> Attachments { get; set; }
        public Organsation Organisation {get;set;}
        public string TermAndCondition { get; set; }
    }
    public class QuoteStatus
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Count { get; set; }
    }
    public class QuoteItem
    {
        public long Id { get; set; }
        public long QuoteId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public double? DiscountRate { get; set; }
        public long? TaxId { get; set; }
        public decimal LineTotal { get; set; }
        public decimal LineTotalIncludeTax { get; set; }
        public string Description { get; set; }
        public long ProductId { get; set; }
        public long? LocationId { get; set; }

        public virtual Location Location { get; set; }
        public virtual Product Product { get; set; }
        public virtual Tax Tax { get; set; }
    }

    public class QuoteForUpdateOrCreate
    {
        public Quote Quote { get; set; }
        public List<Currency> Currencies { get; set; }
        public List<Location> Locations { get; set; }
        public List<Tax> Taxes { get; set; }
        public int BaseCurrencyId { get; set; }
        public Currency TaxCurrency { get; set; }
        public string Number { get; set; }

    }
}
